diff --git a/modules/field.views.inc b/modules/field.views.inc
index dd4363f..3ba8012 100644
--- a/modules/field.views.inc
+++ b/modules/field.views.inc
@@ -101,8 +101,14 @@
 
   $data = array();
 
-  $current_table = _field_sql_storage_tablename($field);
-  $revision_table = _field_sql_storage_revision_tablename($field);
+  // Use the storage details on the field object to work out out the current
+  // table and revision table instead of assuming the field_sql_storage engine
+  //reset($field['storage']['details']);
+  $storage_detail_key = key($field['storage']['details']);
+  //reset($field['storage']['details'][$storage_detail_key]["FIELD_LOAD_CURRENT"]);
+  $current_table = key($field['storage']['details'][$storage_detail_key]["FIELD_LOAD_CURRENT"]);
+  //reset($field['storage']['details'][$storage_detail_key]["FIELD_LOAD_REVISION"]);
+  $revision_table = key($field['storage']['details'][$storage_detail_key]["FIELD_LOAD_REVISION"]);
 
   // The list of entity:bundle that this field is used in.
   $bundles_names = array();
@@ -296,10 +302,10 @@
       else {
         $group = t('@group (historical data)', array('@group' => $group_name));
       }
-      $column_real_name = $field['storage']['details']['sql'][$type][$table][$column];
+      $column_real_name = $field['storage']['details'][$storage_detail_key][$type][$table][$column];
 
       // Load all the fields from the table by default.
-      $additional_fields = array_values($field['storage']['details']['sql'][$type][$table]);
+      $additional_fields = array_values($field['storage']['details'][$storage_detail_key][$type][$table]);
 
       $data[$table][$column_real_name] = array(
         'group' => $group,
